### ISSUE TITLE
Separate stdout/stderr when run docker exec

### ISSUE THREAD

**Issue by philpep** [docker/docker-py#704]
Hi, I would like to known how to distinguish stdout and stderr when exec inside a running container through docker-py:

``` python
>>> client = docker.Client()
>>> container = client.create_container("ubuntu", cmd="tail -f /dev/null")
>>> client.start(container)
>>> e = client.exec_create(container, ["ls", "-l", "/root", "/nonexistent"])
>>> client.exec_start(e)
'ls: cannot access /nonexistent: No such file or directory\n/root:\ntotal 0\n'
>>> # Stdout and stderr are mixed here
```

**Comment by shin-** [comment]
Hm, if you only need one or the other, you can set it up at the `exec_create` level by setting `stderr` or `stdout` to `False` respectively. Outside of that, as far as I know and as far as the [remote API reference](http://docs.docker.com/docker/reference/api/docker_remote_api_v1.20/) is concerned, there's no way to filter the logs of an exec instance.

**Comment by goodwillcoding** [comment]
Are there any plans to address this?

**Comment by shin-** [comment]
I'm not privy to the plans of the engine team, so you'll have to ask [those guys](https://github.com/docker/docker) if an attach-like functionality is planned for `exec` instances. =/

https://github.com/docker/docker/issues/9527 seems like a good start.

**Comment by philpep** [comment]
The strange thing is that the docker native client can do that:

```console
$ id=$(docker run -d ubuntu tail -f /dev/null)
$ docker exec $id ls -l /root /nonexistent 2>/tmp/stderr.txt; echo "STDERR: "; cat /tmp/stderr.txt
/root:
total 0
STDERR: 
ls: cannot access /nonexistent: No such file or directory
```

Docker logs during exec:

```
time="2015-08-12T10:47:32.499394595+02:00" level=debug msg="Calling POST /containers/{name:.*}/exec"
time="2015-08-12T10:47:32.499433254+02:00" level=info msg="POST /v1.20/containers/92b53d25e274fa78a538678be432702a308246b4af7f10a1ff98a3acf5dd4a7e/exec"
time="2015-08-12T10:47:32.499964918+02:00" level=debug msg="Calling POST /exec/{name:.*}/start"
time="2015-08-12T10:47:32.499988307+02:00" level=info msg="POST /v1.20/exec/1c13e8b5ca4e3755dc3c776635328c14d2e43806e4caf4b480ad3daa6ac4d121/start"
time="2015-08-12T10:47:32.500051966+02:00" level=debug msg="starting exec command 1c13e8b5ca4e3755dc3c776635328c14d2e43806e4caf4b480ad3daa6ac4d121 in container 92b53d25e274fa78a538678be432702a308246b4af7f10a1ff98a3acf5dd4a7e"
time="2015-08-12T10:47:32.500113023+02:00" level=debug msg="attach: stdout: begin"
time="2015-08-12T10:47:32.500136084+02:00" level=debug msg="attach: stderr: begin"
time="2015-08-12T10:47:32.631466826+02:00" level=debug msg="Exec task in container 92b53d25e274fa78a538678be432702a308246b4af7f10a1ff98a3acf5dd4a7e exited with code 2"
time="2015-08-12T10:47:32.631522181+02:00" level=debug msg="attach: stdout: end" 
time="2015-08-12T10:47:32.631553231+02:00" level=debug msg="attach: stderr: end" 
2015/08/12 10:47:32 http: response.WriteHeader on hijacked connection
time="2015-08-12T10:47:32.631810063+02:00" level=debug msg="Calling GET /exec/{id:.*}/json"
time="2015-08-12T10:47:32.631823120+02:00" level=info msg="GET /v1.20/exec/1c13e8b5ca4e3755dc3c776635328c14d2e43806e4caf4b480ad3daa6ac4d121/json"
```

I don't known how to inspect the complete POST and GET requests.

**Comment by sharkspeed** [comment]
had anyone fixed this ?

**Comment by Darnok99** [comment]
I'm also interested in getting that separated. Shell way of doing things is to put everything you want other people to see in stdout and all of the debugging in the stderr. With docker run we can only get either mixed output or each of streams in separate

**Comment by shin-** [comment]
Resolved in #2150
